src/lib/translators/mongo.ts

```diff
@@ -527,6 +527,91 @@ function transformSplit(step: Readonly<S.SplitStep>): MongoStep {
   ];
 }
 
+/** transform a 'statistics' step into corresponding mongo steps */
+function transformStatistics(step: Readonly<S.StatisticsStep>): MongoStep {
+  // Some useful formula:
+  const quintilesIndexFormula: any = {
+    maxIndex: { $subtract: ['$count', 1] },
+    lastQuintileIndex: { $multiply: [{ $divide: ['$count', 10] }, 9] },
+    lastQuartileIndex: { $multiply: [{ $divide: ['$count', 4] }, 3] },
+    medianIndex: { $divide: ['$count', 2] },
+    firstQuartileIndex: { $divide: ['$count', 4] },
+    firstQuintileIndex: { $divide: ['$count', 10] },
+    minIndex: 0,
+  };
+  const varianceFormula: any = { $subtract: ['$average_sum_square', { $pow: ['$average', 2] }] };
+  const otherStatisticsFormula: any = {
+    count: 1,
+    average: 1,
+    variance: varianceFormula,
+    std: {
+      // std = square_root(N/(N-1) * variance) (With N = count) to get the unbiased version of std
+      $pow: [
+        { $multiply: [{ $divide: ['$count', { $subtract: ['$count', 1] }] }, varianceFormula] },
+        0.5,
+      ],
+    },
+  };
+  const isQuintileStatistics = (statistic: string): boolean =>
+    Object.keys(quintilesIndexFormula).includes(`${statistic}Index`);
+  // ---
+  // Some explicit conditions to compute average or columns square:
+  const computeColumnSquare = (step: Readonly<S.StatisticsStep>): boolean =>
+    step.statistics.includes('variance') || step.statistics.includes('std');
+  const computeAverage = (step: Readonly<S.StatisticsStep>): boolean =>
+    step.statistics.includes('average') ||
+    step.statistics.includes('variance') ||
+    step.statistics.includes('std');
+  // ---
+
+  const stepQuantileIndex: any = { $addFields: {} };
+  let statistic: string;
+  for (statistic of step.statistics) {
+    if (isQuintileStatistics(statistic)) {
+      stepQuantileIndex.$addFields[`${statistic}Index`] = {
+        $trunc: quintilesIndexFormula[`${statistic}Index`], //TODO: it would be more canonical to use a round version of index here. Migrating to Mongo 4.2 will allow use to use the $round step.
```

Comment:
Actually `$trunc` is good for what you want to do. I don't understand the comment ?

Reply:
It is actually related to your comment here:https://github.com/ToucanToco/weaverbird/pull/522#discussion_r422598341

I used a definition of quantile where their values are actual values in the data: `p-th quantile of order n is the value of the data which is above p/n% of other values and below (1-p/n)% of other values`.

 For instance, in the dataset [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11] the median is `5` (the value which "half of the data is smaller and at most and half is greater") not 5.5 (which is not a value inside the data).

With that definition, using `$trunc` or `$round` is the same for median, but for other quantiles it can end to a different result. With the dataset [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11] the 9th decile is `10` (with my definition of quantile: ) but with truncated index we get `9`...